Smol bugfix for Honduras and even smaller change for Turkey

dogukans1 opened the pull request:
As for Honduras:
1. "Support the USA" focus code is improved to make sure Honduras is joining USA against Fascist Germany, and accounted the case where USA takes over Allies. This focus' description explicitly states that Honduras wishes to join USA that fights German Reich
2. Added code to remove "Monroe Doctrine" spirit for every focus that used to remove US guarantee as US no longer simulates that by guarantees but rather a spirit

As for Turkey: "Modest Economic Liberalisation" focus is locked if Etaism focus is completed. This focus makes Celal Bayar the PM, which locks all paths associated to Etaism and costs at least 210 PP to get rid of him with a suitable PM

dogukans1 committed:
Update turkey.txt

Added "not completed Contiune of the policy of Etatism" check to prevent the paths under etatism locked because they need different PM's than Celal Bayar

I know you can replace him, but it takes 200PP (to remove and add another PM that allows you to contiune your path)

dogukans1 committed:
Update turkey.txt

dogukans1 committed:
Update turkey.txt

dogukans1 committed:
Update honduras.txt

marijn211 commented:
Although the focus description may target Germany, we prefer the focus to be flexible

dogukans1 committed:
Update honduras.txt

dogukans1 commented:
Reverted to the old state. Added bypass checks in case UK and/or US is no longer Democratic so a Democratic Honduras enjoyer doesn't get themselves locked on an ahistorcial game

marijn211 committed:
Merge branch 'Beta-Edits' into pr/2972

marijn211 reviewed (changes requested):
Fixing some typos and seeing if the focus is flexible enough, then accepting

marijn211 committed:
Enhance bugfix
